[PROTO-1489] [PROTO-1512] Add ddex queue flows with db persistence

theoilie opened the pull request:
### Description
Sets up the flow:
1. Uploaded XML or ZIP (now supported in this PR) file goes into postgres. Queue periodically fetches unprocessed XML from postgres and parses them into releases.
2. Another queue periodically checks postgres for new releases and uploads them to Audius when their scheduled release date is met.

Major TODOs:
- Parsing has a ton of work to do.
- We'll need endpoints+UI to display the status of each ZIP, XML, and release.

### How Has This Been Tested?
1. `npm run dev:stage` in packages/ddex
2. `npm run start:stage` in packages/ddex-frontend, and upload an example XML file
3. Backend shows it going through the queue steps and reaching the upload stage. It hangs at `[audius-sdk][tracks-api] Parsing inputs`

michellebrier commented on packages/ddex/src/services/xmlProcessorService.ts:
```diff
@@ -0,0 +1,71 @@
+      for (const row of pending) {
+        // TODO: If we do this then if the server restarts it'll see everything as processing and not re-add it to the queue.
+        // Need to add all 'processing' to the queue on server start in case it restarted in the middle of processing.
+        await sql`UPDATE xml_files SET status = 'processing' WHERE id = ${row.id}`
```
why separate `addXmlFile` and `enqueuePendingXml` rather than adding to the queue in `addXmlFile`?

theoilie replied:
i was keeping the structure the same as for releases where it has to go into the db first and then fetched later when the release_date is reached. but yeah for this i might as well just enqueue as soon as added to the db

michellebrier commented on packages/ddex/package.json:
```diff
@@ -18,13 +18,16 @@
     "postgres": "3.4.3"
   },
   "devDependencies": {
+    "@types/decompress": "^4.2.7",
```
nit: pin this

michellebrier reviewed (approved):
looks good to me as long as we keep the db schema TBD as we figure out parsing

theoilie commented:
@michellebrier i made your requested changes and some fixes to make the flow work (now it reaches the end and the SDK upload is what hangs):
- changed the way it parses XML to work with NodeJS (the previous method that worked in the browser was hanging in Node)
- made it reset the db on each restart
- linted

this worked locally without uncommenting anything about CORS btw (just the 2 steps from the testing section of the PR). also i changed the upload route to use in-memory storage for multer so there's no uploads/ folder to worry about

michellebrier commented:
> @michellebrier i made your requested changes and some fixes to make the flow work (now it reaches the end and the SDK upload is what hangs):
> 
> * changed the way it parses XML to work with NodeJS (the previous method that worked in the browser was hanging in Node)
> * made it reset the db on each restart
> * linted
> 
> this worked locally without uncommenting anything about CORS btw (just the 2 steps from the testing section of the PR). also i changed the upload route to use in-memory storage for multer so there's no uploads/ folder to worry about

sounds good! you can remove the .gitignore in ddex/ since we dont have the ddex/uploads/ folder anymore
